src/unix/bsd/freebsdlike/freebsd/mod.rs

```diff
@@ -1355,6 +1357,66 @@ s! {
         pub strchange_instrms: u16,
         pub strchange_outstrms: u16,
     }
+
+    pub struct filedesc {
+        pub fd_files: *mut fdescenttbl,
+        pub fd_map: *mut c_void,
+        pub fd_freefile: c_int,
+        pub fd_refcnt: c_int,
+        pub fd_holdcnt: c_int,
+        pub fd_sx: sx,
+        pub fd_kqlist: kqlist,
+        pub fd_holdleaderscount: c_int,
+        pub fd_holdleaderswakeup: c_int,
+    }
+
+    pub struct fdescenttbl {
+        pub fdt_nfiles: c_int,
```

Comment:
So all I need is this file. But if I understand correctly, I still need `filedescent` definition because of the `fdt_ofiles` field.

Reply:
I think you can define fdt_ofiles as a `[*mut c_void; 0]` instead to get the correct alignment.

Reply:
If so that'd make things MUCH simpler. Let's give it a try.